app/timeline-menu/timeline-menu.ts

```diff
@@ -0,0 +1,63 @@
+let _ = require('lodash');
+
+export enum TimelineMenuRange { LastWeek, LastMonth, LastYear, LastDecade, PreDecade, CustomRange }
+
+export class TimelineMenu {
+  LIMIT_NUM = 8640000000000000;
+  MIN_DATE = new Date(-this.LIMIT_NUM);
+  MAX_DATE = new Date(this.LIMIT_NUM);
+  periods = [
+    { 'title': '7 ימים אחרונים', 'value': 'LastWeek', 'start': '', 'end': '' },
+    { 'title': '30 ימים אחרונים', 'value': 'LastMonth', 'start': '', 'end': '' },
+    { 'title': 'השנה', 'value': 'LastYear', 'start': '', 'end': '' },
+    { 'title': 'בעשור האחרון', 'value': 'LastDecade', 'start': '', 'end': '' },
+    { 'title': 'מלפני עשור ויותר', 'value': 'PreDecade', 'start': '', 'end': '' },
+    { 'title': 'טווח תאריכים לבחירתך', 'value': 'CustomRange', 'start': '', 'end': '' }
+  ];
+
+  constructor() {
+    this.initPeriods();
+  }
+
+  getPeriod(range: TimelineMenuRange) {
+    return _.find(this.periods, ['value', TimelineMenuRange[range]]);
```

Comment:
not very efficient...
isn't it better to have a map with the names of the ranges as keys?

Reply:
the periods is small and static object but generally you are right. Do you have same example of using map with keys?